### PR TITLE
{Role} Fix #26629: `az ad sp delete`: update help

### LINKED ISSUE #26629: az ad sp delete claims to delete role assignments but it doesn't
### Describe the bug

https://learn.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest#az-ad-sp-delete

> az ad sp delete: Delete a service principal and its role assignments.

This command only deletes the service principal but not its role assignments.

This appears to have been an intentional change and is likely just incorrect documentation/help message at this point.

### Related command

az ad sp delete

### Errors

N/A

### Issue script & Debug output

N/A

### Expected behavior

Role assignments are removed when the SP is deleted.

### Environment Summary

```
azure-cli                         2.49.0

core                              2.49.0
telemetry                          1.0.8

Extensions:
ai-examples                        0.2.5
cloud-service                      0.2.0
ml                                2.17.2
reservation                        0.2.0
ssh                                1.1.6

Dependencies:
msal                              1.20.0
azure-mgmt-resource               22.0.0

Python location '/usr/bin/python3.9'
Extensions directory '/home/admin/.azure/cliextensions'
Extensions system directory '/usr/lib/python3.9/site-packages/azure-cli-extensions'

Python (Linux) 3.9.14 (main, Feb  8 2023, 03:39:52) 
[GCC 11.2.0]
```

### Additional context

#20805 indicates that this is intentional
#22432 performed the change
https://learn.microsoft.com/en-us/cli/azure/microsoft-graph-migration#az-ad-sp-delete documents that the behavior was changed.

Both CLI docs at https://learn.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest#az-ad-sp-delete and `az ad sp delete --help` indicate that role assignments would be removed.

### PR DESCRIPTION
**Pull request by MoChilia**
**Related command**

`az ad sp delete`

**Description**

As @Nothing4You reported in #26629, `az ad sp delete` no longer removes the service principal's role assignments after we migrated to Microsoft Graph API. This can be also checked in source code as follows.
https://github.com/Azure/azure-cli/blob/d741c63e0819ddfb91c9ce21b1943ebb5e88b7a9/src/azure-cli/azure/cli/command_modules/role/custom.py#L1081-L1084
https://github.com/Azure/azure-cli/blob/f7479671e841d196d3312523c302bab853eb25c2/src/azure-cli/azure/cli/command_modules/role/_msgrpah/_graph_client.py#L197-L200
Hence, we correct the help explanation in this pr.

Close #26629 

**Testing Guide**

`az ad sp delete -h`

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Role

File: src/azure-cli/azure/cli/command_modules/role/_help.py [hunk 1]
```diff
@@ -560,9 +560,9 @@
 
 helps['ad sp delete'] = """
 type: command
-short-summary: Delete a service principal and its role assignments.
+short-summary: Delete a service principal.
 examples:
-  - name: Delete a service principal and its role assignments. (autogenerated)
+  - name: Delete a service principal. (autogenerated)
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/_help.py:565]: We can delete `(autogenerated)` and L567 once this command has been reviewed by human.

```suggestion
  - name: Delete a service principal.
```

File: src/azure-cli/azure/cli/command_modules/role/_help.py [hunk 2]
```diff
@@ -560,9 +560,9 @@
...
     text: az ad sp delete --id 00000000-0000-0000-0000-000000000000
     crafted: true
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/_help.py:567]:
```suggestion
```

**Comment by jiasli** [comment]
Nice and clear PR description!

**Pull request merged**